### PR TITLE
Add empty files #91

### LINKED ISSUE #91: Initialize empty files from within drive
Some things to consider:
* What types of files do we want to support? Potential answer: All file types recognized as Text/Code
* Where do we initialize files? Probably in the same dialog as the 'add directory'

Other than that it should be straight-forward.

### PR DESCRIPTION
**Pull request by matheus23**
Closes #91

Code should be ready for review, but there's some polishing on the UX side to do:
* [x] Empty the text field on successful creation
* [x] Prevent creating the same file twice (at the moment this overrides the file to be empty, yikes)
* [x] Give the user some visual feedback for clicking the buttons
* [ ] ~~Open newly created text files with an editor~~ Postponed for now

### CODE REVIEW

File: src/Application/Drive/View/Sidebar.elm [hunk 1]
```diff
@@ -88,7 +88,7 @@ viewSidebar { scrollable, expanded, body } =
 
         --
         , if scrollable then
-            T.overflow_y_scroll
+            T.overflow_y_auto
```
Comment (icidasset) [inline src/Application/Drive/View/Sidebar.elm:91]: 👍

File: src/Application/Radix.elm [hunk 2]
```diff
@@ -101,10 +101,10 @@ type Msg
     | CloseSidebar
     | CopyPublicUrl { item : Item, presentable : Bool }
     | CopyToClipboard { clip : String, notification : String }
-    | CreateDirectory
     | DigDeeper { directoryName : String }
     | DownloadItem Item
-    | GotCreateDirectoryInput String
+    | GotAddOrCreateInput String
+    | CreateFileOrFolder (Maybe { extension : String })
```
Comment (icidasset) [inline src/Application/Radix.elm:107]: Nitpick, can you put this alphabetical order 🙏 
Same for `State.elm`

**Review by icidasset — APPROVED** [review]
One small note, looks good otherwise 👏

**Comment by matheus23** [comment]
I decided to skip _actually_ figuring out when the files/folders were created. This requires some coordination from Elm -> Ports -> Webnative -> Ports -> Elm, and I'm not quite sure how to do this without refactoring a little.

How do the toasty notifications for removing/renaming files work @icidasset ? They seem to be a good starting point, but all I could find was `model.fileSystemStatus` and `Toasty.addConditionalToast`. From looking at the code it seems to me like when I quickly remove lots of files, as soon as `model.fileSystemStatus` goes to `Ready` once, they all should disappear, but that's not the case.

I have some ideas on how to do this Elm -> Webnative -> Elm communication, which might also help reduce the amount of code in `FileSystem.State#gotDirectoryList`, but I want to talk about that with you before I try implementing that.

**Comment by icidasset** [comment]
> This requires some coordination from Elm -> Ports -> Webnative -> Ports -> Elm

Yeah that's not that easy, for sure. I was thinking of doing this https://discourse.elm-lang.org/t/service-worker-ffi/6408 pattern, so we can use `Task`s for our webnative interactions (and thus chain filesystem actions). But then we're depending on service workers for the most crucial parts of the system... which is probably not a good idea? 🤔 

> From looking at the code it seems to me like when I quickly remove lots of files, as soon as model.fileSystemStatus goes to Ready once, they all should disappear, but that's not the case.

It's not that advanced no. The condition is just for adding the notification on that exact moment. Which in some cases doesn't even make sense, because it's always going to be true. That library I used doesn't have that much features, can't even remove a notification, so we should probably look for something better.

> I have some ideas on how to do this Elm -> Webnative -> Elm communication

Shoot 🔫

**Review by icidasset — APPROVED** [review]

**Pull request merged**